src/main/java/net/helpscout/api/utils/HTTPConnectionUtils.java

```diff
@@ -0,0 +1,144 @@
+
+        conn.setRequestProperty("Authorization", "Basic " + EncodeUtils.getEncoded(apiKey + ":x"));
+        conn.setRequestProperty("Accept-Encoding", "gzip, deflate");
+        if (requestBody != null) {
+            conn.setRequestProperty("Content-Type", "application/json");
+            conn.setRequestProperty("Accept", "application/json");
+            conn.setDoOutput(true);
+            try (OutputStream output = conn.getOutputStream()) {
+                output.write(requestBody.getBytes("UTF-8"));
+            }
+        }
+        conn.connect();
+        checkStatusCode(conn, expectedCode);
+        return conn;
+    }
+
+    private static void checkStatusCode(HttpURLConnection conn, int expectedCode) throws ApiException, IOException {
+        int code = conn.getResponseCode();
+
+        if (code == expectedCode) {
+            return;
+        }
+
+        switch(code) {
+            case HTTP_BAD_REQUEST:
+                String details = getDetailedErrorMessage(conn);
+                throw new InvalidFormatException("The request was not formatted correctly", details);
+            case HTTP_UNAUTHORIZED:
+                throw new InvalidApiKeyException("Invalid API key");
+            case HTTP_PAYMENT_REQUIRED:
+                throw new ApiKeySuspendedException("API key suspended");
+            case HTTP_FORBIDDEN:
+                throw new AccessDeniedException("Access denied");
+            case HTTP_NOT_FOUND:
+                throw new NotFoundException("Resource not found");
+            case HTTP_BAD_METHOD:
+                throw new InvalidMethodException("Invalid method type");
+            case HTTP_REQUESTS_LIMIT:
+                throw new ThrottleRateException("Throttle limit reached. Too many requests");
+            case HTTP_INTERNAL_ERROR:
+                throw new ServerException("Application error or server error", getDetailedErrorMessage(conn));
+            case HTTP_UNAVAILABLE:
+                throw new ServiceUnavailableException("Service Temporarily Unavailable");
+            default:
+                throw new ApiException("Unknown API exception");
+        }
+    }
+
+    private static String getDetailedErrorMessage(HttpURLConnection conn) throws IOException {
+        InputStream is = conn.getErrorStream();
+        String json = "";
+        if (is != null) {
+            json = IOUtils.toString(is, "UTF-8");
+        }
+
+        return StringUtils.isNotEmpty(json) ? new JsonFormatter().format(json) : null;
+    }
+
+    public static void close(HttpURLConnection conn) {
+        if (conn != null) {
+            try {
+                conn.disconnect();
+            } catch (Exception e) {
+                // ignore
+            }
+        }
+    }
+
+    public static String getResponse(HttpURLConnection conn) throws IOException {
+
+        StringBuilder sb = new StringBuilder();
+
+        try (BufferedReader br = new BufferedReader(new InputStreamReader(
```

Comment:
This might get rid of lot of code - https://commons.apache.org/proper/commons-io/javadocs/api-1.4/org/apache/commons/io/IOUtils.html#toString(java.io.InputStream)